### PR TITLE
[Reader Improvements] Post Details bottom actions bar

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #19227 

This PR implements the updates to the bottom actions bar inside the Post Details screen to match the new specs. 

The only possibly missing part is updating the icon assets to match the specs (zQnohyMpLzBzQ5jzMMKni3-fi-869:20299#564589863) but we need @osullivanchris to check if that's needed and provide some missing assets in that case. 

**Note:** We will proceed with the code review and merge of this PR even without the confirmation about the icons since this is pointing to the feature branch and we can update it in another PR.

This change is also behind the `ReaderImprovementsFeatureConfig` flag.

## Demo
Flag OFF | Flag ON
--- | ---
image | image 

## Test
1. Install Jetpack app and login
2. Turn the `ReaderImprovementsFeatureConfig` config off
3. Go to the Reader
4. Open a Post
5. **Verify** the bottom bar in the Post Details still matches the old design
6. **Verify** the bottom bar actions work as expected
7. Turn the `ReaderImprovementsFeatureConfig` config on
3. Go to the Reader
4. Open a Post
5. **Verify** the bottom bar in the Post Details matches the new design
6. **Verify** the bottom bar actions work as expected

## Regression Notes
1. Potential unintended areas of impact
Affect similar action bars in other parts of the app (such as the posts list).

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt [hunk 1]
```diff
@@ -459,14 +466,32 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
...
+                    ReaderIncludePostDetailFooterBinding.bind(inflated).mapBinding()
+                }
+
+                val elevationOverlayProvider = ElevationOverlayProvider(layoutFooterBinding.root.context)
+                val appbarElevation = resources.getDimension(R.dimen.appbar_elevation)
+                val elevatedSurfaceColor = elevationOverlayProvider.compositeOverlayWithThemeSurfaceColorIfNeeded(
+                    appbarElevation
+                )
+                layoutFooterBinding.root.setBackgroundColor(elevatedSurfaceColor)
+                layoutFooterBinding.root.visibility = View.INVISIBLE
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt]: Nitpick: Would it make sense to use the `isInvisible` extension function from `androidx.core.view.View` instead?
e.g.
```
layoutFooterBinding.root.isInvisible = true
```

**Review by RenanLukas — APPROVED** [review]
Thanks for the PR, @thomashorta . It's working as expected and design looks good :shipit:

**Comment by osullivanchris** [comment]
Looks good. In the screenshot it looks a bit cramped, but feels/looks ok on my device. A couple of possible changes to discuss:

- in dark mode, is it necessary for the bottom action sheet to take the raised background color? Could it just be on black? I think it takes too much attention on dark mode with the different background color to the rest of the page.
- Is it possible for the bar to hide on scroll?
- Do you think its better to hide or grey out comment/like when they are unavailable?

**Comment by thomashorta** [comment]
> * in dark mode, is it necessary for the bottom action sheet to take the raised background color? Could it just be on black? I think it takes too much attention on dark mode with the different background color to the rest of the page.

It should be possible. Do you want a solid black in that case or the current `surface` color (#121212).

Surface Color (#121212) | Black (#000000)
--- | ---
details-footer-surface | details-footer-black

> * Is it possible for the bar to hide on scroll?

How would the behavior be here though? I mean, when would the bar reappear?

> * Do you think its better to hide or grey out comment/like when they are unavailable?

I think either way could work fine, though I prefer disabling (greying out) as the UI remains consistent and makes it clearer that that action can be available in other posts/blogs.

**Comment by osullivanchris** [comment]
> It should be possible. Do you want a solid black in that case or the current surface color (#121212).

I think the surface color, so it just blends in a little more. What do you think of it?

> How would the behavior be here though? I mean, when would the bar reappear?

On Reader and Notifications top level, the navigation bar hides on scroll, shows on scroll back. I don't know if there's anywhere we're doing it for bottom tab bars, but I think the same principle would work. Facebook on iOS does this. As it happens I am not finding examples of Android apps with the bottom tab bars hiding on scroll, only top navigation bars and top tabs. so perhaps it is more challenging?

> I think either way could work fine, though I prefer disabling (greying out) as the UI remains consistent and makes it clearer that that action can be available in other posts/blogs.

ok agree. I think on the feed cards hiding them to save space makes sense, and the actions are not in a very memorable position. Whereas here the tap targets can become quite memorable. so its nice to keep them in the same position 👍

**Comment by thomashorta** [comment]
> I think the surface color, so it just blends in a little more. What do you think of it?

I agree, I think the surface color looks better!

> As it happens I am not finding examples of Android apps with the bottom tab bars hiding on scroll, only top navigation bars and top tabs. so perhaps it is more challenging?

I found it on X (Twitter). I did a quick search and found out that the Material lib provides an off-the-shelf bottom bar behavior for that which is literally just adding 1 line of code and it worked pretty well. I liked this addition!

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/fbf3aaca-21be-4804-802f-0a9c528b9dee

> I think on the feed cards hiding them to save space makes sense, and the actions are not in a very memorable position. Whereas here the tap targets can become quite memorable. so its nice to keep them in the same position

Exactly, I didn't know how to explain my preference but that's exactly it! Thanks! 🙇🏼

**Comment by osullivanchris** [comment]
Nice work @thomashorta glad we are in agreement on those details 👍  That hiding of the bottom bar looks great! If its just as easy to hide the top navigation bar too I would do it to let all the focus be on reading the article.

**Comment by thomashorta** [comment]
> If its just as easy to hide the top navigation bar too I would do it to let all the focus be on reading the article.

Is this what you had in mind? If so, yes, it is easy to implement, though it's a bit harder to implement it just for the Reader Improvements Feature Flag. Are you OK with having this behavior on the top bar even on the old layout? This would only be visible to the user in the old layout in case we ever release the changes with the Feature Flag OFF or decide to turn it off for some reason, which I think is unlikely, but asking anyway.

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/b3f91e81-bbc5-4e90-ba5d-5bad32bf8504

**Comment by osullivanchris** [comment]
> Is this what you had in mind? If so, yes, it is easy to implement, though it's a bit harder to implement it just for the Reader Improvements Feature Flag. Are you OK with having this behavior on the top bar even on the old layout? This would only be visible to the user in the old layout in case we ever release the changes with the Feature Flag OFF or decide to turn it off for some reason, which I think is unlikely, but asking anyway.

Yep thats fine I think its an improvement in the old and in the new, and it is low risk 👍

**Comment by thomashorta** [comment]
Just pushed some final commit based on the review comments from both @RenanLukas and @osullivanchris. It should be ready for another round of review.

**Review by RenanLukas — APPROVED** [review]
Thanks for the changes. LGTM :shipit:

**Pull request merged**